### PR TITLE
endpoint task its check for available urls and it filters inappropriate urls

### PR DESCRIPTION
**Pull request by qnislavnachev**

### CODE REVIEW

**Comment by EmilGeorgiev** [comment]
```Endpoint Task is done and tested``` is bad name for PR. From this name and description i can not understand what is functionality or task which is implemented.

File: src/testwithmockendpointtask/StartsWithKeyword.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package testwithmockendpointtask;
+
+public class StartsWithKeyword implements Endpoint {
+    @Override
+    public boolean matches(String url) {
+        String keyword = "www";
```
Comment (EmilGeorgiev) [inline src/testwithmockendpointtask/StartsWithKeyword.java]: this string is better to be added as parameter in constructor. here if i change ```www``` to ```ccc``` tests will fail. Here you test state not behavior. State is this string ```www``` or digest ```5```, ```6``` and etc.

https://dzone.com/articles/test-behavior-not-state

Comment (denisdimitrov) [inline src/testwithmockendpointtask/StartsWithKeyword.java]: I understand. I will change it.

File: test/testwithmockendpointtask/EndpointFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        context.checking(new Expectations() {{
+            oneOf(endpoint).matches(url);
+            will(returnValue(true));
+        }});
+        assertTrue(f.shouldFilter(url));
+    }
+
+    @Test
+    public void startWithKeyword() throws Exception {
+        StartsWithKeyword matcher = new StartsWithKeyword("www");
```
Comment (EmilGeorgiev) [inline test/testwithmockendpointtask/EndpointFilterTest.java]: you should have different test classes for ```Endpoint``` and ```StartWithKeyword```

**Comment by EmilGeorgiev** [comment]
LGTM after fix comment

**Pull request closed without merging**